zerver/tests/test_custom_profile_data.py

```diff
@@ -751,6 +740,66 @@ def test_removing_option_from_select_field(self) -> None:
             CustomProfileFieldValue.objects.filter(field_id=field.id, value="1").exists()
         )
 
+    def test_external_account_type_field(self) -> None:
+        self.login("iago")
+        realm = get_realm("zulip")
+        field_data = orjson.dumps(
+            {
+                "subtype": "twitter",
+            }
+        ).decode()
+
+        field = CustomProfileField.objects.get(name="GitHub", realm=realm)
+
+        # Subtype of external account type profile field cannot be updated.
+        result = self.client_patch(
+            f"/json/realm/profile_fields/{field.id}",
+            info={
+                "name": "GitHub",
+                "field_type": CustomProfileField.EXTERNAL_ACCOUNT,
+                "field_data": field_data,
+            },
+        )
+        self.assert_json_error(result, "External account type cannot be updated.")
+
+        field_data = orjson.dumps(
+            {
+                "subtype": "github",
+            }
+        ).decode()
+
+        result = self.client_patch(
+            f"/json/realm/profile_fields/{field.id}",
+            info={
+                "name": "GitHub",
+                "field_type": CustomProfileField.EXTERNAL_ACCOUNT,
+                "field_data": field_data,
+            },
+        )
+        self.assert_json_success(result)
```

Comment:
What's the purpose of this test?

Reply:
I am attempting to change the "subtype" field here